File: source/com/barrybecker4/simulation/reactiondiffusion/rendering/RDRenderingOptions.java
/** Copyright by Barry G. Becker, 2000-2011. Licensed under MIT License: http://www.opensource.org/licenses/MIT  */
package com.barrybecker4.simulation.reactiondiffusion.rendering;

import com.barrybecker4.common.concurrency.RunnableParallelizer;

/**
 * Date: Aug 15, 2010
 *
 * @author Barry Becker
 */
public class RDRenderingOptions {

    private boolean isShowingU = false;
    private boolean isShowingV = true;

    /** used for scaling the bump height. if 0, then no bumpiness. */
    private double heightScale = 0;

    /** Specular highlight degree. */
    private double specularConst = 0;

    /** Manages the worker threads. */
    private RunnableParallelizer parallelizer;


    /**
     * Constructor
     */
    public RDRenderingOptions() {
        setParallelized(false);
    }

    public void setParallelized(boolean useParallelization) {

        parallelizer =
             useParallelization ? new RunnableParallelizer() : new RunnableParallelizer(1);
    }

    RunnableParallelizer getParallelizer() {
        return parallelizer;
    }

    public boolean isParallelized() {
        return parallelizer.getNumThreads() > 1;
    }

    public void setHeightScale(double h) {
        heightScale = h;
    }

    protected double getHeightScale() {
        return heightScale;
    }


    public void setSpecular(double s) {
        specularConst = s;
    }

    public double getSpecular() {
        return specularConst;
    }

    public boolean isShowingU() {
        return isShowingU;
    }

    public void setShowingU(boolean showingU) {
        isShowingU = showingU;
    }


    public boolean isShowingV() {
        return isShowingV;
    }

    public void setShowingV(boolean showingV) {
        isShowingV = showingV;
    }
}

File: scala-source/com/barrybecker4/simulation/reactiondiffusion/rendering/RDRenderingOptions.scala
/** Copyright by Barry G. Becker, 2000-2017. Licensed under MIT License: http://www.opensource.org/licenses/MIT  */
package com.barrybecker4.simulation.reactiondiffusion.rendering

/**
  * Rendering specific options
  *
  * @author Barry Becker
  */
class RDRenderingOptions() {
  setParallelized(false)
  private var showingU: Boolean = false
  private var showingV:Boolean = true
  /** used for scaling the bump height. if 0, then no bumpiness. */
  private var heightScale: Double = 0
  /** Specular highlight degree. */
  private var specularConst: Double = 0
  private var runParallelized = false

  def setParallelized(useParallelization: Boolean) { runParallelized = useParallelization}
  def isParallelized: Boolean = runParallelized

  def setHeightScale(h: Double) { heightScale = h }
  def getHeightScale: Double = heightScale

  def setSpecular(s: Double) { specularConst = s}
  def getSpecular: Double = specularConst

  def isShowingU: Boolean = this.showingU
  def setShowingU(showingU: Boolean) {this.showingU = showingU}

  def isShowingV: Boolean = showingV
  def setShowingV(showingV: Boolean) { this.showingV = showingV}
}